src/test/unit_tests/test_s3_ops.js

```diff
@@ -464,7 +476,13 @@ mocha.describe('s3_ops', function() {
             await s3.getObject({ Bucket: bucket_name, Key: text_file2 }).promise();
             await s3.getObject({ Bucket: bucket_name, Key: text_file3 }).promise();
 
+            assert.strictEqual(res1.Contents[0].Key, text_file1);
+            assert.strictEqual(res1.Contents[0].Size, file_body.length);
+            assert.strictEqual(res1.Contents.length, 3);
+
+            if (!caching) return;
             const query_params = { get_from_cache: true };
+            await P.delay(200);
```

Comment:
as we are trying not to use bluebird, I think we can change that to:
```suggestion
            const delay = util.promisify(setTimeout);
            await delay(null, 200);
```
@guymguym what do you think?